src/ir/gc-type-utils.h

```diff
@@ -95,6 +95,10 @@ inline EvaluationResult evaluateCastCheck(Type refType, Type castType) {
   if (refIsHeapSubType) {
     assert(refType.isNullable());
     assert(castType.isNonNullable());
+    if (refHeapType.isBottom()) {
+      // Non-null references to bottom types do not exist!
+      return Failure;
```

Comment:
The comment on line 88 needs to be updated (or maybe just removed), since we can no longer say at that point that it will not definitely fail.